data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/Sink.java

```diff
@@ -20,7 +20,7 @@
      *
      * @param records the records to write to the sink.
      */
-    void output(Collection<T> records);
+    void output(Collection<T> records, boolean copyFlag);
```

Comment:
I don't think this is quite the right design. It is putting in a value for the `PipelineConnector` which other sinks will not need.

The `PipelineConnector` is a unique Sink and is internal to Data Prepper core.

Is there another design which can accomplish this goal without affecting the `Sink` interface or any other sinks?